cli/command/environment.go

```diff
@@ -13,5 +18,250 @@ func NewEnvironmentCommand(b *CommandBase) *EnvironmentCommand {
+		EnvironmentName:  args["NAME"],
+		InstanceSize:     c.String("size"),
+		MinClusterCount:  c.Int("min-count"),
+		UserDataTemplate: userData,
+		OperatingSystem:  c.String("os"),
+		AMIID:            c.String("ami"),
+	}
+
+	jobID, err := e.client.CreateEnvironment(req)
+	if err != nil {
+		return err
+	}
+
+	return e.waitOnJobHelper(c, jobID, "creating", func(environmentID string) error {
+		environment, err := e.client.ReadEnvironment(environmentID)
+		if err != nil {
+			return err
+		}
+
+		return e.printer.PrintEnvironments(environment)
+	})
+}
+
+func (e *EnvironmentCommand) delete(c *cli.Context) error {
+	return e.deleteHelper(c, "environment", func(environmentID string) (string, error) {
+		return e.client.DeleteEnvironment(environmentID)
+	})
+}
+
+func (e *EnvironmentCommand) list(c *cli.Context) error {
+	environmentSummaries, err := e.client.ListEnvironments()
+	if err != nil {
+		return err
+	}
+
+	return e.printer.PrintEnvironmentSummaries(environmentSummaries...)
+}
+
+func (e *EnvironmentCommand) read(c *cli.Context) error {
+	args, err := extractArgs(c.Args(), "NAME")
+	if err != nil {
+		return err
+	}
+
+	environmentIDs, err := e.resolver.Resolve("environment", args["NAME"])
+	if err != nil {
+		return err
+	}
+
+	environments := make([]*models.Environment, len(environmentIDs))
+	for i, environmentID := range environmentIDs {
+		environment, err := e.client.ReadEnvironment(environmentID)
+		if err != nil {
+			return err
+		}
+
+		environments[i] = environment
+	}
+
+	return e.printer.PrintEnvironments(environments...)
+}
+
+func (e *EnvironmentCommand) update(c *cli.Context) error {
+	args, err := extractArgs(c.Args(), "NAME", "COUNT")
+	if err != nil {
+		return err
+	}
+
+	minClusterCount, err := strconv.Atoi(args["COUNT"])
+	if err != nil {
+		return fmt.Errorf("'%s' is not a valid integer", args["COUNT"])
+	}
+
+	id, err := e.resolveSingleEntityIDHelper("environment", args["NAME"])
+	if err != nil {
+		return err
+	}
+
+	req := models.UpdateEnvironmentRequest{
+		EnvironmentID:   id,
+		MinClusterCount: &minClusterCount,
+	}
+
+	jobID, err := e.client.UpdateEnvironment(req)
+	if err != nil {
+		return err
+	}
+
+	return e.waitOnJobHelper(c, jobID, "updating", func(environmentID string) error {
+		environment, err := e.client.ReadEnvironment(environmentID)
+		if err != nil {
+			return err
+		}
+
+		return e.printer.PrintEnvironments(environment)
+	})
+}
+
+func (e *EnvironmentCommand) link(c *cli.Context) error {
+	args, err := extractArgs(c.Args(), "SOURCE", "DESTINATION")
+	if err != nil {
+		return err
+	}
+
+	sourceID, err := e.resolveSingleEntityIDHelper("environment", args["SOURCE"])
+	if err != nil {
+		return err
+	}
+
+	destID, err := e.resolveSingleEntityIDHelper("environment", args["DESTINATION"])
+	if err != nil {
+		return err
+	}
+
+	if sourceID == destID {
+		return fmt.Errorf("Cannot link an environment to itself")
+	}
+
+	if err := e.client.CreateLink(sourceID, destID); err != nil {
```

Comment:
this should be using the job model - bit since we have changes incoming there’s no reason to fix it here. Just something to remember for future